### PR TITLE
fix (api): more precision in the log

### PR DESCRIPTION
**Pull request by yesnault**
There is something strange,

```
Run> Unable to load to pipeline scheduler 150 pq: could not obtain lock on row in relation "pipeline_scheduler"
```

This error should be "55P03", so ignored from log.Error. This commit is to have the real pqerr.Code

Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

File: engine/api/error.go [hunk 1]
```diff
@@ -10,8 +10,15 @@ import (
...
 func WriteError(w http.ResponseWriter, r *http.Request, err error) {
 	al := r.Header.Get("Accept-Language")
-	msg, code := sdk.ProcessError(err, al)
+	msg, errProcessed := sdk.ProcessError(err, al)
 	sdkErr := sdk.Error{Message: msg}
-	log.Warning("%-7s | %-4d | %s \t %s", r.Method, code, r.RequestURI, err)
-	WriteJSON(w, r, sdkErr, code)
+
+	// ErrAlreadyTaken is not useful to log in warning
+	if errProcessed.ID == sdk.ErrAlreadyTaken.ID {
```
Comment (fsamin) [inline engine/api/error.go]: sdk.ErrorIs ?

File: engine/api/error.go [hunk 2]
```diff
@@ -10,8 +10,15 @@ import (
...
+		log.Debug("%-7s | %-4d | %s \t %s", r.Method, errProcessed.Status, r.RequestURI, err)
+	} else {
+		log.Warning("%-7s | %-4d | %s \t %s", r.Method, errProcessed.Status, r.RequestURI, err)
```
Comment (fsamin) [inline engine/api/error.go]: I find it really interesting to see that there is a lot of contention on every jobs, It means we still have things to improve.
If you are upset with the warning, the log .Info would be fine for me if we are currently using the info log-level

Comment (yesnault) [reply engine/api/error.go]: yes, Info

File: engine/api/error.go (hunk 1 above)
Comment (yesnault) [reply engine/api/error.go]: ? (I don't understand the comment :) )

Comment (yesnault) [reply engine/api/error.go]: hum, ok. errorIs(...), I'll fix

**Pull request merged**